database/index.go

```diff
@@ -136,25 +183,88 @@ func (idx *Index) Delete(v document.Value, k []byte) error {
 	return engine.ErrKeyNotFound
 }
 
+// validatePivots returns an error when the pivots are unsuitable for the index:
+// - no pivots at all
+// - having pivots length superior to the index arity
+// - having the first pivot without a value when the subsequent ones do have values
+func (idx *Index) validatePivots(pivots []document.Value) {
+	if len(pivots) == 0 {
+		panic("cannot iterate without a pivot")
```

Comment:
🕐 We might want to fall in cases where there are no pivot but we want to read from the index anyway:

```sql
SELECT * FROM foo ORDER BY a [ASC|DESC]
```

If `a` is indexed, this will probably use the index to avoid having to sort the table (which is memory intensive). In this case, we won't have any range but will use the index to iterate over everything.